dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeMetadataConnector.java

```diff
@@ -148,16 +152,18 @@ protected void addSqlTasksWithInfoSchemaFallback(
     }
   }
 
-  @ForOverride
-  protected void addSingleSqlTask(
+  protected void addSingleSqlTaskForAssessment(
       @Nonnull List<? super Task<?>> out,
       @Nonnull Class<? extends Enum<?>> header,
```

Comment:
[New review] Per discussion: Suggest to make this CheckForNull, avoid passing it when possible, do **NOT** pass it in any query that looks like a 'select *' (e.g. show warehouses is really a select * from ???), and let's not add the additional complexity.